File: Selenium/Activities/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-controls")
    print(f"The page title is : {driver.title}")
	
    explicitwait  = WebDriverWait(driver,10)
    print(f"The page title is : {driver.title}")
	
    chkBox = driver.find_element(By.XPATH,"//input[@id = 'checkbox']")
    print(f"Is the checkbox visible on the page ? : {chkBox.is_displayed()}")
    btnToggle =  driver.find_element(By.XPATH,"//button[text() = 'Toggle Checkbox']")
    btnToggle.click()
    explicitwait.until(EC.invisibility_of_element(chkBox))
    print(f"Is the checkbox visible on the page ? After clicking on toggle button : {chkBox.is_displayed()}")
			
    btnToggle.click()
    explicitwait.until(EC.element_to_be_clickable(chkBox)).click()
			
    print(f"Toggle button is clicked again to make the Checkbox is visible . Checkbox is selected : {chkBox.is_selected()}")
    driver.quit()

File: Selenium/Activities/Activity8.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity8 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			driver.get("https://training-support.net/webelements/dynamic-controls");
			WebDriverWait explicitwait  =  new WebDriverWait(driver,Duration.ofSeconds(10));
			
			System.out.println("The page title is : "+ driver.getTitle());
			
			WebElement chkBox = driver.findElement(By.xpath("//input[@id = 'checkbox']"));
			System.out.println("Is the checkbox visible on the page ? : " + chkBox.isDisplayed());
			WebElement btnToggle =  driver.findElement(By.xpath("//button[text() = 'Toggle Checkbox']"));
			btnToggle.click();
			
			explicitwait.until(ExpectedConditions.invisibilityOf(chkBox));
			System.out.println("Is the checkbox visible on the page ? After clicking on toggle button : " + chkBox.isDisplayed());
			
			btnToggle.click();
			explicitwait.until(ExpectedConditions.elementToBeClickable(chkBox)).click();
			
			System.out.println("Toggle button is clicked again to make the Checkbox is visible . Checkbox is selected : " + chkBox.isSelected());
			
			driver.quit();

	}

}
